api_app/script_analyzers/observable_analyzers/urlscan.py

```diff
@@ -0,0 +1,80 @@
+import requests
+import time
+from api_app.exceptions import AnalyzerRunException
+from api_app.script_analyzers import classes
+from intel_owl import secrets
+
+
+class UrlScan(classes.ObservableAnalyzer):
+    base_url: str = "https://urlscan.io/"
+
+    def set_config(self, additional_config_params):
+        self.analysis_type = additional_config_params.get("urlscan_analysis", "search")
+        api_key_name = additional_config_params.get("api_key_name", "URLSCAN_KEY")
+        self.visibility = additional_config_params.get("visibility", "public")
+        self.__api_key = secrets.get_secret(api_key_name)
+
+    def __submit_query(self):
+        data = {"url": self.observable_name, "visibility": self.visibility}
+        uri = "api/v1/scan/"
+        try:
+            response = requests.post(
+                self.base_url + uri, headers=self.headers, json=data
+            )
+            response.raise_for_status()
+        except requests.RequestException as e:
+            raise AnalyzerRunException(e)
+        return response.json()["api"]
+
+    def __poll_for_result(self, url, headers):
+        max_tries: int = 10
+        poll_distance: int = 2
+        result = {}
+        for chance in range(max_tries):
+            time.sleep(poll_distance)
+            resp = requests.get(url, headers=headers)
+            resp_json = resp.json()
+            status = resp_json.get("status", None)
+            if status == 404:
```

Comment:
this is still wrong. you have to check for the status code, not the "status" keyword.

Reply:
Status code is accessible via `resp.status_code`